storage/beacondb/kvstore.go

```diff
@@ -115,7 +115,9 @@ func New(kss store.KVStoreService) *KVStore {
 			schemaBuilder,
 			sdkcollections.NewPrefix([]byte{keys.ForkPrefix}),
 			keys.ForkPrefixHumanReadable,
-			encoding.SSZValueCodec[*ctypes.Fork]{},
+			encoding.SSZValueCodec[*ctypes.Fork]{
+				NewEmptyF: ctypes.NewEmptyFork,
```

Comment:
this is my proposal to avoid all the `Interface[Self T]` madness.
Just provide the empty constructor where needed and that will take care of instantiating the object where needed.
On the contrary with the `pointer.New` structure, we have to worry whether `pointer` is nil or not and what is the semantics of `New` (returns a different pointer? overwrites `pointer`? Both?)

Maybe not a big deal, but I fill simple things should be simple